### PR TITLE
Make aggregation of distributed results extensible

### PR DESCRIPTION
**Pull request by mrcslws**
Enables mixins to handle aggregation.

In a future PR I'm adding a mixin that saves the training loss for every batch and stores it as a list in the result dict. The current PR allows me to aggregate those lists.

### CODE REVIEW

File: nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py [hunk 1]
```diff
@@ -195,13 +192,13 @@ def _create_workers(self, config):
 
         self._process_config(config)
 
-        experiment_class = config["experiment_class"]
+        self.experiment_class = config["experiment_class"]
```
Comment (lscheinkman) [inline nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py:195]: I would initialize this field at the beginning of `_setup` method instead `_create_workers` method

Comment (mrcslws) [reply nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py:195]: Good point.
